lib/Elastica/JSON.php

```diff
@@ -45,17 +46,42 @@ public static function parse(/* inherit from json_decode */)
      * Convert input to JSON string with standard options.
      *
      * @link http://php.net/manual/en/function.json-encode.php
+     * @link http://php.net/manual/en/function.json-last-error.php
+     *
+     * @param mixed $args,... Target to stringify
      *
-     * @param  mixed check args for PHP function json_encode
+     * @throws JSONParseException
      *
      * @return string Valid JSON representation of $input
      */
-    public static function stringify(/* inherit from json_encode */)
+    public static function stringify($args/* inherit from json_encode */)
     {
         // extract arguments
         $args = func_get_args();
 
         // run encode and output
-        return call_user_func_array('json_encode', $args);
+        $string = call_user_func_array('json_encode', $args);
+
+        // turn errors into exceptions for easier catching
+        if ($error = self::getJsonLastErrorMsg()) {
+            throw new JSONParseException($error);
+        }
+
+        // output
+        return $string;
+    }
+
+    /**
+     * Get Json Last Error
+     *
+     * @link http://php.net/manual/en/function.json-last-error.php
+     * @link http://php.net/manual/en/function.json-last-error-msg.php
+     * @link https://github.com/php/php-src/blob/master/ext/json/json.c#L308
+     *
+     * @return string
+     */
+    private static function getJsonLastErrorMsg()
+    {
+        return JSON_ERROR_NONE !== json_last_error() ? json_last_error_msg() : false;
```

Comment:
It's a little bit inconsistent that the method can return string and bool, but it is PHP, so fair enough ;-)